sunpy/map/mapbase.py

```diff
@@ -2739,6 +2739,14 @@ def reproject_to(self, target_wcs, *, algorithm='interpolation', return_footprin
         outmap = GenericMap(output_array, target_wcs.to_header(),
                             plot_settings=self.plot_settings)
 
+        # Check rsun mismatch
+        rsun_map = self.meta["rsun_ref"]
+        rsun_target = outmap.meta["rsun_ref"]
+        if rsun_map != rsun_target:
```

Comment:
Would the walrus operator be useful here?

Reply:
Did you mean using something like this??
```
if (rsun_map := self.meta["rsun_ref"]) != (rsun_target := outmap.meta["rsun_ref"]):
```

Reply:
Yeah, would that work?

Reply:
Yes

Reply:
I see that pre-commit isn’t happy about local variables `rsun_map` and `rsun_target` which are never use. 
Instead can we simply do this 
```
if self.meta["rsun_ref"] != outmap.meta["rsun_ref"]:
```
as there is no real use of assigning `self.meta["rsun_ref"]` `outmap.meta["rsun_ref"]` to local variables.

Reply:
Were they not used for the error message?

Reply:
No. For the error message we make direct calls to `self` and `outmap`.
```python
warn_user("rsun mismatch detected: "
         f"{self.name}.rsun_obs={self.rsun_obs}; {outmap.name}.rsun_obs={outmap.rsun_obs}. "
```

Reply:
Ah I misread the warn statement.

Reply:
but then why can not we use the same keywords here instead of defining new variables?

Reply:
Yes we can simply do something like this:
```python
if self.rsun_meters != outmap.rsun_meters:
   warn_user("rsun mismatch detected: "
            f"{self.name}.rsun_meters={self.rsun_meters}; {outmap.name}.rsun_meters={outmap.rsun_meters}. "
```
Or

Reply:
I would assume so if that is what we are using to check the difference between two maps?